File: apps/Offline-AI-Buddy/android/core/src/main/java/com/americangroupllc/offlineaibuddy/core/monetization/EntitlementService.kt
package com.americangroupllc.offlineaibuddy.core.monetization

import com.americangroupllc.offlineaibuddy.core.models.EntitlementState

/**
 * Subscription/lifetime IAP entitlement gate. Mirrors
 * `BuddyAICore.EntitlementService`.
 */
interface EntitlementService {
    suspend fun currentEntitlement(): EntitlementState
    suspend fun purchaseSubscription(productId: String)
    suspend fun purchaseLifetime(productId: String)
    suspend fun restorePurchases()
}

class NoopEntitlementService : EntitlementService {
    override suspend fun currentEntitlement(): EntitlementState = EntitlementState.FREE
    override suspend fun purchaseSubscription(productId: String) = Unit
    override suspend fun purchaseLifetime(productId: String) = Unit
    override suspend fun restorePurchases() = Unit
}

interface AdGate {
    suspend fun isReady(): Boolean
    /** Present a cached interstitial. Returns true if the user finished it. */
    suspend fun watchAd(): Boolean
}

class NoopAdGate : AdGate {
    override suspend fun isReady(): Boolean = false
    override suspend fun watchAd(): Boolean = true   // Dev-build pretend
}

File: apps/Offline-AI-Buddy/shared/BuddyAICore/Sources/BuddyAICore/Monetization/EntitlementService.swift
import Foundation

/// Slot for the subscription / lifetime IAP entitlement. v1 ships a
/// RevenueCat-backed impl AND a `NoopEntitlementService` that always
/// returns the free tier (used in dev builds without the RevenueCat
/// API key set).
public protocol EntitlementService: Sendable {
    func currentEntitlement() async -> EntitlementState
    /// Open the platform paywall + complete purchase. Updates entitlement
    /// state on success. Throws on user cancel.
    func purchaseSubscription(productId: String) async throws
    func purchaseLifetime(productId: String) async throws
    func restorePurchases() async throws
}

public final class NoopEntitlementService: EntitlementService {
    public init() {}

    public func currentEntitlement() async -> EntitlementState { .free }

    public func purchaseSubscription(productId: String) async throws {
        // No-op in dev builds.
    }

    public func purchaseLifetime(productId: String) async throws {
        // No-op in dev builds.
    }

    public func restorePurchases() async throws {
        // No-op in dev builds.
    }
}

#if canImport(RevenueCat)
// import RevenueCat
// public final class RevenueCatEntitlementService: EntitlementService { ... }
// — wired in iOS app target (not in shared package) so the SPM dep
//   stays out of the test target.
#endif
